src/Runtime/Runtime/System.Windows/UIElement.cs

```diff
@@ -1745,10 +1745,33 @@ private void BeginUpdateCustomLayout()
             FrameworkElement fe = this as FrameworkElement;
             if (fe != null)
             {
+                double offsetLeft;
+                double offsetTop;
+                if (CSharpXamlForHtml5.Environment.IsRunningInJavaScript)
+                {
+                    // Browser
+                    var boundingClientRect = ((dynamic)INTERNAL_OuterDomElement).getBoundingClientRect();
+                    offsetLeft = boundingClientRect.left - boundingClientRect.left;
+                    offsetTop = boundingClientRect.top - boundingClientRect.top;
+                }
+                else
+                {
+                    // Simulator
+                    string boundingClientRect = Convert.ToString(OpenSilver.Interop.ExecuteJavaScript(
+                        "$0.getBoundingClientRect().left + '|' + $0.getBoundingClientRect().top",
+                        INTERNAL_OuterDomElement));
+                    offsetLeft = Convert.ToDouble(boundingClientRect.Split('|')[0], CultureInfo.InvariantCulture);
+                    offsetTop = Convert.ToDouble(boundingClientRect.Split('|')[1], CultureInfo.InvariantCulture);
+                }
+
                 if (fe.IsAutoWidthOnCustomLayout)
-                    layoutMeasuredSize.Width = double.PositiveInfinity;
+                {
+                    layoutMeasuredSize.Width = Math.Min(Window.Current.ActualWidth - offsetLeft, fe.MaxWidth);
```

Comment:
MeasureCore will make sure width is smaller than MaxWidth
```suggestion
                    layoutMeasuredSize.Width = Window.Current.ActualWidth - offsetLeft;
```

Reply:
Got it. In the end, this change has been reverted entirely, as a premature return on PopupHelper was the main cause of the issue (not limiting the popup height). Thank you!